modules/consentManagementGpp.js

```diff
@@ -0,0 +1,430 @@
+    return;
+  }
+
+  const callbacks = {
+    onSuccess: (data) => done(data, false),
+    onError: function (msg, ...extraArgs) {
+      done(null, true, msg, ...extraArgs);
+    }
+  }
+  cmpCallMap[userCMP](callbacks);
+
+  if (!isDone) {
+    const onTimeout = () => {
+      const continueToAuction = (data) => {
+        done(data, false, 'GPP CMP did not load, continuing auction...');
+      }
+      processCmpData(provisionalConsent, {
+        onSuccess: continueToAuction,
+        onError: () => continueToAuction(storeConsentData(undefined))
+      })
+    }
+    if (consentTimeout === 0) {
+      onTimeout();
+    } else {
+      timer = setTimeout(onTimeout, consentTimeout);
+    }
+  }
+}
+
+/**
+ * Like `loadConsentData`, but cache and re-use previously loaded data.
+ * @param cb
+ */
+function loadIfMissing(cb) {
+  if (consentData) {
+    logInfo('User consent information already known.  Pulling internally stored information...');
+    // eslint-disable-next-line standard/no-callback-literal
+    cb(false);
+  } else {
+    loadConsentData(cb);
+  }
+}
+
+/**
+ * If consentManagement module is enabled (ie included in setConfig), this hook function will attempt to fetch the
+ * user's encoded consent string from the supported CMP.  Once obtained, the module will store this
+ * data as part of a gppConsent object which gets transferred to adapterManager's gppDataHandler object.
+ * This information is later added into the bidRequest object for any supported adapters to read/pass along to their system.
+ * @param {object} reqBidsConfigObj required; This is the same param that's used in pbjs.requestBids.
+ * @param {function} fn required; The next function in the chain, used by hook.js
+ */
+export const requestBidsHook = timedAuctionHook('gpp', function requestBidsHook(fn, reqBidsConfigObj) {
+  loadIfMissing(function (shouldCancelAuction, errMsg, ...extraArgs) {
+    if (errMsg) {
+      let log = logWarn;
+      if (shouldCancelAuction) {
+        log = logError;
+        errMsg = `${errMsg} Canceling auction as per consentManagement config.`;
+      }
+      log(errMsg, ...extraArgs);
+    }
+
+    if (shouldCancelAuction) {
+      fn.stopTiming();
+      if (typeof reqBidsConfigObj.bidsBackHandler === 'function') {
+        reqBidsConfigObj.bidsBackHandler();
+      } else {
+        logError('Error executing bidsBackHandler');
+      }
+    } else {
+      fn.call(this, reqBidsConfigObj);
+    }
+  });
+});
+
+/**
+ * This function checks the consent data provided by CMP to ensure it's in an expected state.
+ * If it's bad, we call `onError`
+ * If it's good, then we store the value and call `onSuccess`
+ */
+function processCmpData(consentObject, {onSuccess, onError}) {
+  function checkData() {
+    const gppString = consentObject && consentObject.gppString;
+    const gppSection = (consentObject && consentObject.applicableSections) ? consentObject.applicableSections
```

Comment:
checkApplicableSectionsIsReady() and checkApplicableSectionIsReady() can be used to be consistent on how the existing sections are verified

Reply:
I can move those functions into here as well.